docs: update pull_request.md and PULL_REQUEST_TEMPLATE.md (#6106)
codeigniter4/CodeIgniter4: .github/PULL_REQUEST_TEMPLATE.md

Each pull request should address a single issue and have a meaningful title.

- All bug fixes should be sent to the __"develop"__ branch, this is where the next bug fix version will be developed.

> - PRs with any enhancement should be sent to __"4.3"__ branch, this is where the next minor upgrade version will be developed.

Reviewer: Can we make this generic so we don't have to keep updating the document?
```suggestion
- PRs with any enhancement should be sent to the next minor version branch, e.g. __"4.3"__
```
